openwisp_controller/geo/api/serializers.py

```diff
@@ -38,3 +35,209 @@ class Meta:
         model = Location
         geo_field = 'geometry'
         fields = '__all__'
+
+
+class BaseSerializer(FilterSerializerByOrgManaged, ValidatedModelSerializer):
+    pass
+
+
+class FloorPlanSerializer(BaseSerializer):
+    class Meta:
+        model = FloorPlan
+        fields = (
+            'id',
+            'floor',
+            'image',
+            'location',
+            'created',
+            'modified',
+        )
+        read_only_fields = ('created', 'modified')
+
+    def validate(self, data):
+        if data.get('location'):
+            data['organization'] = data.get('location').organization
+        instance = self.instance or self.Meta.model(**data)
+        instance.full_clean()
+        return data
+
+
+class FloorPlanLocationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = FloorPlan
+        fields = (
+            'floor',
+            'image',
+        )
+        extra_kwargs = {'floor': {'required': False}, 'image': {'required': False}}
+
+
+class LocationSerializer(FilterSerializerByOrgManaged, serializers.ModelSerializer):
```

Comment:
I will change `serializers.ModelSerializer` to `gis_serializers.GeoFeatureModelSerializer` once the testing is done.